ondir.c

```diff
@@ -171,6 +175,8 @@ const char *src = NULL, *dst = NULL, *home = NULL;
 		if (dst[len] == '/') ++len;
 	}
 
+	if (dst && dst != cwd)
+		free((void*)dst);
```

Comment:
other lines are added because of the strdup(), which is there to avoid modifying buffers that may not be owned by the application.